bot/exts/moderation/voice_gate.py

```diff
@@ -160,6 +186,35 @@ async def on_message(self, message: discord.Message) -> None:
         with suppress(discord.NotFound):
             await message.delete()
 
+    @Cog.listener()
+    async def on_voice_state_update(self, member: Member, *_) -> None:
+        """Pings a user if they've never joined the voice chat before and aren't verified."""
+        if member.bot:
+            log.trace("User is a bot. Ignore.")
+            return
+
+        in_cache = await self.redis_cache.get(member.id, None)
+
+        # member.voice will return None if the user is not in a voice channel
+        if in_cache is None and member.voice is not None:
+            log.trace("User not in cache and is in a voice channel")
+            verified = any(Roles.voice_verified == role.id for role in member.roles)
+            if verified:
+                log.trace("User is verified, add to the cache and ignore")
+                #  redis cache does not accept None, so False is used to signify no message
+                await self.redis_cache.set(member.id, False)
+                return
+
+            log.trace("User is unverified. Send ping.")
+            message = await self._voice_verification_channel.send(VOICE_PING.format(member.mention))
+            await self.redis_cache.set(member.id, message.id)
```

Comment:
This races with `_async_init`. There is no guarantee that the task has completed, so the attr may not yet exist. You can await the task, or simply get rid of `_async_init`, await the guild ready event here, and grab a fresh instance every time. I also think that attrs that are not set in the `__init__` should be explicitly annotated or assigned a placeholder `None` at init time, so that you don't have to worry about the potential `AttributeError`

Reply:
Corrected in the latest commit.